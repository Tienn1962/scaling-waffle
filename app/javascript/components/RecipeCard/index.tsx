import React from "react";
import {
  ClockIcon,
  ShoppingCartIcon,
  StarIcon,
  UserIcon,
} from "@heroicons/react/outline";

interface Props {
  title: string;
  ratings: number;
  prepTime: number;
  cookTime: number;
  ingredients: string[];
  image_url?: string;
  author?: string;
}

export default (props: Props) => {
  return (
    <div className="w-60 card bg-neutral text-neutral-content">
      <figure>
        <img
          className="object-fit h-60"
          src={props.image_url || "assets/No-Image-Placeholder.png"}
          alt={props.title}
        />
      </figure>
      <div className="card-body">
        <h2 className="card-title">{props.title}</h2>
        {props.author && (
          <div className="flex">
            <UserIcon className="h-6 w-6" />
            <span className="pl-2">{props.author}</span>
          </div>
        )}
        <div className="flex">
          <ClockIcon className="h-6 w-6" />
          <span className="pl-2">
            {props.prepTime}m | {props.cookTime}m
          </span>
        </div>
        <div className="flex">
          <StarIcon className="h-6 w-6" />
          <span className="pl-2">{props.ratings}/5</span>
        </div>
        <div className="flex">
          <ShoppingCartIcon className="h-6 w-6" />
          <span className="pl-2">{props.ingredients.length} ingredients</span>
        </div>
      </div>
    </div>
  );
};
