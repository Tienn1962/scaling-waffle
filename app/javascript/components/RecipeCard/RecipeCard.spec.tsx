import React from "react";
import { render } from "@testing-library/react";
import RecipeCard from ".";

it("renders a card", () => {
  const component = render(
    <RecipeCard
      title="The spiciest potato"
      ratings={4.9}
      prepTime={30}
      cookTime={30}
      ingredients={["1 potato"]}
    />
  );

  expect(component).toMatchSnapshot();
});
