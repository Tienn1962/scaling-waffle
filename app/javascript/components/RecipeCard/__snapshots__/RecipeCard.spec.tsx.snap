// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a card 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-60 card bg-neutral text-neutral-content"
      >
        <figure>
          <img
            alt="The spiciest potato"
            class="object-fit h-60"
            src="assets/No-Image-Placeholder.png"
          />
        </figure>
        <div
          class="card-body"
        >
          <h2
            class="card-title"
          >
            The spiciest potato
          </h2>
          <div
            class="flex"
          >
            <svg
              aria-hidden="true"
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <span
              class="pl-2"
            >
              30
              m | 
              30
              m
            </span>
          </div>
          <div
            class="flex"
          >
            <svg
              aria-hidden="true"
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <span
              class="pl-2"
            >
              4.9
              /5
            </span>
          </div>
          <div
            class="flex"
          >
            <svg
              aria-hidden="true"
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <span
              class="pl-2"
            >
              1
               ingredients
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-60 card bg-neutral text-neutral-content"
    >
      <figure>
        <img
          alt="The spiciest potato"
          class="object-fit h-60"
          src="assets/No-Image-Placeholder.png"
        />
      </figure>
      <div
        class="card-body"
      >
        <h2
          class="card-title"
        >
          The spiciest potato
        </h2>
        <div
          class="flex"
        >
          <svg
            aria-hidden="true"
            class="h-6 w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="pl-2"
          >
            30
            m | 
            30
            m
          </span>
        </div>
        <div
          class="flex"
        >
          <svg
            aria-hidden="true"
            class="h-6 w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="pl-2"
          >
            4.9
            /5
          </span>
        </div>
        <div
          class="flex"
        >
          <svg
            aria-hidden="true"
            class="h-6 w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="pl-2"
          >
            1
             ingredients
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
